modules/statistics/src/main/java/io/uhndata/cards/statistics/StatisticQueryServlet.java

```diff
@@ -0,0 +1,503 @@
+     */
+    protected Map<String, String> parseArguments(SlingHttpServletRequest request) throws IOException
+    {
+        JsonParser parser = Json.createParser(request.getInputStream());
+        Map<String, String> retVal = new HashMap<>();
+        while (parser.hasNext()) {
+            Event event = parser.next();
+            if (event == JsonParser.Event.KEY_NAME) {
+                String key = parser.getString();
+                event = parser.next();
+                retVal.put(key, parser.getString());
+            }
+        }
+
+        return retVal;
+    }
+
+    /**
+     * Split: Get all answers that have a given question filled out.
+     * @param data the map that the return values will be added to
+     * @param type given type (x or split)
+     * @param question The question node that the answers is to
+     * @param resolver Reference to the resource resolver
+     * @return map containing all question nodes and their given type
+     */
+    private Map<Resource, String> getAnswersWithType(Map<Resource, String> data, String type, Node question,
+        ResourceResolver resolver) throws RepositoryException
+    {
+        final StringBuilder query =
+            // We select all answers that answer our question
+            new StringBuilder("select n from [cards:Answer] as n where n.'question'='"
+                + question.getIdentifier() + "'");
+        Iterator<Resource> answers = resolver.findResources(query.toString(), Query.JCR_SQL2);
+
+        while (answers.hasNext()) {
+            Resource answer = answers.next();
+            data.put(answer, type);
+        }
+
+        return data;
+    }
+
+    /**
+     * Split: Filter the given iterator of resources to only include resources whose parent is a Form, whose
+     * Subject's type is equal to the given subjectType.
+     * @param data Iterator of resources
+     * @param subjectType Subject type of the subject for the answer's form
+     * @return The filtered iterator
+     */
+    private Map<String, Map<Resource, String>> filterAnswersWithType(Map<Resource, String> data, Node subjectType)
+        throws RepositoryException
+    {
+        Iterator<Map.Entry<Resource, String>> entries = data.entrySet().iterator();
+
+        Map<String, Map<Resource, String>> newData = new HashMap<>();
+
+        String correctType = subjectType.getIdentifier();
+
+        // filter out answers without correct subject type
+        while (entries.hasNext()) {
+            Map.Entry<Resource, String> answer = entries.next();
+            // get form node
+            Node formNode = getParentNode(answer.getKey().adaptTo(Node.class));
+            // get parent subject
+            Node formSubject = formNode.getProperty("subject").getNode();
+
+            boolean foundAnswer = false;
+            while (formSubject.getDepth() > 0) {
+                if (formSubject.getProperty("type").getNode().getIdentifier().equals(correctType)) {
```

Comment:
This fails on `formSubject.getProperty("type")` if we don't find the correct type and go too far up the tree, since the subjects homepage does not have that property.

```suggestion
                if (formSubject.hasProperty("type")
                    && formSubject.getProperty("type").getNode().getIdentifier().equals(correctType)) {
```